anchore_engine/services/policy_engine/__init__.py

```diff
@@ -373,8 +379,8 @@ def handle_grypedb_sync(*args, **kwargs):
 
             if result:
                 logger.info("Grype DB synced to local instance via handler")
-        except Exception as e:
-            logger.error("Caught escaped error in grype db sync: {}".format(e))
+        except GrypeDBSyncError:
+            logger.exception("Caught escaped error in grype db sync")
```

Comment:
This is an excellent improvement. Now that you are catching something very explicit, you have the opportunity to expand on the error or simply mention that this may happen. Do you see value in having the full traceback logged? The message here is not meaningful, because the logging module will produce the traceback and the exception, so it is implied that there was an error and that the logging module showed what that was, making the message redundant.

Reply:
Upon some further testing and research I think this actually might need to catch all exceptions. The way that these async monitors work is that the infinite loop with the sleep causes the handler to run at every interval defined in the cycle timer. When the function exits, the monitor sees this and then calls the function again immediately. As a result, if a non GrypeDBSyncError occurs regularly, then the monitor will call this function every second and it will error out on every call, flooding the logs and breaking the interval in which this task should run.

Reply:
That sounds like a problem the _monitor_ should handle better. In what situation a non-GrypeDBSyncError would happen regularly? 
If this is catching every single exception and then cycling through, that seems problematic too (although more spaced out). 
You are describing valid concerns, but these concerns look more like a symptom of a handler that is causing broad exception handling.

Reply:
Its not specifically that I believe any other errors will be thrown. I do control for downstream errors in the grype facade by catching those errors and raising them as a GrypeDBSyncError, but its more just an issue with if I don't catch one then it breaks the cycle timer in which the task is supposed to run and instead the monitor continuously calls this handler as I described above.

Reply:
I spoke to @nightfurys about this issue and she said that there is a higher risk of not catching all the errors here than there is with catching all because of how this will behave. This becomes more difficult to address without addressing potential problems with exception handling/raising downstream. The problem is that while some of the code written for these features might be an improvement on exception handling and raising, there is also other code that gets called that needs to be fixed.  Swathi thought that for the moment we should keep with the pattern of catching all exceptions here. I will also note that the scoping of some of the downstream exceptions are likely to change as the facade is further developed. This might not reflect the final scoping that gets merged into master so I can add a TODO to address this again once the downstream exceptions are further fleshed out. I already plan on scoping further down the exception handling in the update_grype_db function

Reply:
I think that adding an issue to describe what needs to happen to address this problem would be a solid way of making progress with this code as-is. The `#TODO` can then reference that issue so that others aren't surprised to see more try/except/Exception added in new code.

Reply:
Created this ticket and linked it to the TODO https://github.com/anchore/anchore-engine/issues/1005